taskw/warrior.py

```diff
@@ -32,9 +33,36 @@
 logger = logging.getLogger(__name__)
 
 
-# Location of configuration file: either specified by TASKRC environment
-# variable, or ~/.taskrc (default).
-TASKRC = os.getenv("TASKRC", "~/.taskrc")
+def find_taskrc():
+    """
+    Find the location of the taskwarrior configuration file.
+
+    First checks ${TASKRC} environment variable, then falls back to ~/.taskrc and finally ${XDG_CONFIG_DIR}/task/taskrc.
+
+    Raises FileNotFoundErrors if either
+      * Specified taskrc is not a file
+      * No taskrc was found
+    """
+    if "TASKRC" in os.environ.keys():
+        taskrc = Path(os.environ["TASKRC"])
+        if taskrc.is_file():
+            return taskrc
+        else:
+            raise FileNotFoundError("Environment variable 'TASKRC' did not resolve to a taskrc file")
+
+
+    taskrc = Path.home() / ".taskrc"
+    if taskrc.is_file():
+        return taskrc
+ 
+    if "XDG_CONFIG_HOME" in os.environ.keys():
+        taskrc = Path(os.environ["XDG_CONFIG_HOME"]) / "task/taskrc"
+        if taskrc.is_file():
+            return taskrc
+
+    raise FileNotFoundError("Unable to find taskrc. Set environment variable 'TASKRC=<file>' for a non-standard location")
+    
+
```

Comment:
Any reason not to assign `TASKRC = find_taskrc()` here? Then you wouldn't have to change anything else in this module and we could avoid the extra conditionals in this module.